packages/core-transactions/src/handlers/transaction.ts

```diff
@@ -110,26 +106,50 @@ export abstract class TransactionHandler implements ITransactionHandler {
         this.revertForRecipient(transaction, walletManager);
     }
 
-    protected applyToSender(transaction: Interfaces.ITransaction, walletManager: State.IWalletManager): void {
+    public applyToSender(transaction: Interfaces.ITransaction, walletManager: State.IWalletManager): void {
         const sender: State.IWallet = walletManager.findByPublicKey(transaction.data.senderPublicKey);
-        sender.balance = sender.balance.minus(transaction.data.amount).minus(transaction.data.fee);
-        sender.incrementNonce();
+        const d: Interfaces.ITransactionData = transaction.data;
```

Comment:
```suggestion
        const { data } = transaction;
```

Reply:
With destructuring it should actually be `const { data }: Interfaces.ITransaction = transaction;` but that would defeat one of the purposes of having types:
```
const variable: TypeOfVariable = expression;
```
VS
```
const { variable }: BlahSomeOtherType = expression;
```